contracts/smart-account/interfaces/base/IModuleManager.sol

```diff
@@ -76,6 +76,12 @@ interface IModuleManager {
         uint256 operationLength
     );
 
+    /**
+     * @notice Throws when trying to remove the only enabled module
+     * @param module Module address provided
```

Comment:
Can you please add a note here or in `_disableModule` or both, stating that this does not cover the case where the last module is an "execution" module? If the last module is an execution module, the SA will make it unrecoverable. It's important that everyone understands the implications of removing a module.

We should really think about an identifier for the modules.

Reply:
I made this addition. 
We can introduce self-declaration for the validation modules via ERC165, however, this doesn't solve for cases when malicious module self declares being validation module, while it is not